### DISCUSSION TITLE
Collection Datasource Throws an Unkown Error

### DISCUSSION THREAD

**Discussion by BenOussama180** [Power-Components/livewire-powergrid#776] [Q&A]
This is my datasource : 
![collection_datasource](https://user-images.githubusercontent.com/81111919/172072937-b642118e-9f87-4760-b4e6-0decd61596d0.png)
The table shows up normally, but after fiiltering or doing an action that refresh the table, it shows this error : 
First iit shows a blank livewire error page : 
![image](https://user-images.githubusercontent.com/81111919/172072972-1a94039d-07a6-450e-b55e-7820fc9030f2.png)
And in the console this : 
livewire.js?id=c69d0f2801c01fcf8166:13          POST https://linkpandas.com/livewire/message/customer-links-table 500
iit throws a 500 error message, but i dont know whats the error, this bug only occurs on datasource collection, on builer it works normally , i have powergrid V3

**Comment by BenOussama180** [comment]
any help? :/

**Comment by luanfreitasdev** [comment]
I have no idea yet, why not use builder instead of collection?

**Comment by BenOussama180** [comment]
basically i want to eliminate links duplicates by site , and only showing the cheapest one from the duplicates (  lowest price ) 
i figured that query out with map, but when i use map on return type of Builder it throws an error
u got any idea on how to get same results with Builder ? thank you @luanfreitasdev

**Comment by luanfreitasdev** [comment]
Does this map work if you manipulate the Helpers/Model or in the component?

**Comment by BenOussama180** [comment]
the map works if return type of Datasource is collection, but it throws an error while filtering, if i switch it to Builder i  get this -> 
datasource : 
![image](https://user-images.githubusercontent.com/81111919/172224263-237400cd-9d32-4ec9-9683-aafa1eac7822.png)
Error : 
![image](https://user-images.githubusercontent.com/81111919/172224312-ea6ef955-bb1b-42a5-8836-15d9fc8d7752.png)

**Comment by BenOussama180** [comment]
any idea on how to get same results  ( retreiving only cheapest link by site from duplicates ) with Eloquent\Builder::class ? @luanfreitasdev Thank you for the help

**Comment by luanfreitasdev** [comment]
Have you tried using Join ?

**Comment by BenOussama180** [comment]
No i didnt because i simply couldnt figure it out with join

**Comment by BenOussama180** [comment]
and i dont need to join two tables there, everything i need is in the same table and Model Links

**Comment by luanfreitasdev** [comment]
I believe it is not a bug in the component. Maybe it could be a new feature or something query builder related.
